packages/web3-utils/src/event_emitter.ts

```diff
@@ -0,0 +1,117 @@
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { EventEmitter as EventEmitterAtNode } from 'events';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type Callback = (params: any) => void | Promise<void>;
+
+type EventTargetCallback = (params: CustomEvent) => void;
+
+const wrapFunction =
+	(fn: Callback): EventTargetCallback =>
+	(params: CustomEvent) =>
+		fn(params.detail);
+
+/**
+ * This class copy the behavior of Node.js EventEmitter class.
+ * It is used to provide the same interface for the browser environment.
+ */
+export class EventEmitterAtBrowser extends EventTarget {
+	private _listeners: Record<string, [key: Callback, value: EventTargetCallback][]> = {};
+	private maxListeners = Number.MAX_SAFE_INTEGER;
+
+	public on(eventName: string, fn: Callback) {
+		this.addEventListener(eventName, fn);
+		return this;
+	}
+
+	public once(eventName: string, fn: Callback) {
+		const onceCallback = async (params: Callback) => {
+			await fn(params);
+			this.off(eventName, onceCallback);
```

Comment:
This PR LGTM, only
`codecov/patch` is not passing, coverage should be increased before merging.